src/components/NavBar/index.tsx

```diff
@@ -0,0 +1,118 @@
+'use client';
+
+import { useMemo } from 'react';
+import Image from 'next/image';
+import Link from 'next/link';
+import { usePathname } from 'next/navigation';
+import COLORS from '@/styles/colors';
+import { Flex } from '@/styles/containers';
+import { useAuth } from '@/utils/AuthProvider';
+import { useActiveStatus } from '@/utils/hooks';
+import { useProfile } from '@/utils/ProfileProvider';
+import IJPlogo from '~/public/images/ijp-logo.webp';
+import { ProfileButton, SmallLinkButton } from '../Buttons';
+import * as Styles from './style';
+
+type NavLink = {
+  name: string;
+  path: string;
+  active: boolean;
+};
+
+export default function NavBar() {
+  const profile = useProfile();
+  if (!profile) throw new Error('Profile must be defined.');
+
+  const auth = useAuth();
+  if (!auth) throw new Error('Auth must be defined.');
+
+  const authButtonView = useMemo(() => {
+    if (profile.profileReady && auth.userId)
+      return (
+        <ProfileButton href="/settings">
+          {profile.profileData?.first_name || 'Profile'}
+        </ProfileButton>
+      );
+
+    return (
+      <>
+        <SmallLinkButton
+          $primaryColor={COLORS.blueMid}
+          $secondaryColor={COLORS.blueDark}
+          href="/login"
+        >
+          Log In
+        </SmallLinkButton>
+        <SmallLinkButton
+          $primaryColor="white"
+          $secondaryColor={COLORS.blueDark}
+          href="/signup"
+          style={{ color: COLORS.blueMid }}
+        >
+          Sign Up
+        </SmallLinkButton>
+      </>
+    );
+  }, [profile, auth.userId]);
+
+  const navlinks: NavLink[] = [
+    { name: 'Cases', path: '/cases', active: useActiveStatus('/cases') },
+    {
+      name: 'Limited Case Assignments',
+      path: '/limited-case-assignments',
+      active: useActiveStatus('/limited-case-assignments'),
+    },
+    {
+      name: 'Language Support',
+      path: '/language-support',
+      active: useActiveStatus('/language-support'),
+    },
+  ];
+
+  const renderLink = (link: NavLink) => (
+    <Styles.LinkContainer key={link.path}>
+      <Flex
+        $direction="column"
+        $justify="center"
+        style={{
+          flex: 1,
+        }}
+      >
+        <Styles.NoUnderlineLink href={link.path} $color="white">
+          <Styles.DisplayText $isActive={link.active}>
+            {link.name}
+          </Styles.DisplayText>
+          {link.name}
+        </Styles.NoUnderlineLink>
+      </Flex>
+      <Styles.ActiveUnderline $isActive={link.active} />
+    </Styles.LinkContainer>
+  );
+
+  const currentPath = usePathname();
+  if (currentPath.includes('/onboarding') && !currentPath.includes('/roles')) {
+    return null;
+  }
+
+  return (
+    <Styles.NavBarContainer>
+      <Styles.NavBarSectionDiv>
+        <Link href="/">
+          <Image
+            alt="background"
+            src={IJPlogo.src}
+            placeholder="blur"
+            blurDataURL={IJPlogo.src}
```

Comment:
Is it necessary to have the blurDataURL, if it is just the same as the src?

Reply:
not sure, it seems like `blurDataURL` acts as a placeholder for the actual image (`src`). Here is a screenshot of Next.js' documentation:

<img width="641" alt="image" src="https://github.com/calblueprint/immigration-justice-project/assets/84427964/e8b1ba9b-1536-40d3-987f-8bf640f0e38a">

Should we consider removing the `blurDataURL` placeholder if we're using the same image as the placeholder and it can impact performance since these source images are likely not 10px lol??

Reply:
Wait, based on Next.js' documentation it seems like `blurDataURL` would be automatically populated in this case:

<img width="662" alt="image" src="https://github.com/calblueprint/immigration-justice-project/assets/84427964/1bceb0ef-e801-4a0f-af74-098c02c12513">

We should be fine with removing `blurDataURL` entirely.

Here is a link to a quick video on it as well: https://www.youtube.com/watch?v=Bz3No1RFXWY

Reply:
For it to be automatically populated, we have to pass in `IJPLogo` entirely to `src`. Not just the source path.

Reply:
ah ok so `src={IJPlogo}` and we remove `blurDataURL`? Should we move forward with this for all other images then?

Reply:
If it works, I'd be down! Let's experiment with it a little bit first though, I'm also not entirely sure

Reply:
ah nvm let's keep it as is for now. Here is a video of the effect of applying this change: 

https://github.com/calblueprint/immigration-justice-project/assets/84427964/4ea31d35-a35c-448a-9a14-3380185467cc